hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/reader/attributes/ValueSource.java

```diff
@@ -1,28 +1,19 @@
 package org.hypertrace.trace.reader.attributes;
 
 import java.util.Optional;
+import org.apache.avro.generic.GenericRecord;
 import org.hypertrace.core.attribute.service.v1.AttributeKind;
 import org.hypertrace.core.attribute.service.v1.LiteralValue;
-import org.hypertrace.core.datamodel.Event;
-import org.hypertrace.core.datamodel.StructuredTrace;
 import org.hypertrace.core.grpcutils.client.rx.GrpcRxExecutionContext;
 
-interface ValueSource {
+public interface ValueSource<T extends GenericRecord, S extends GenericRecord> {
```

Comment:
What is value source using its type parameters for? It looks like it's just reflecting them

Reply:
Yeah. I started with adding generics, because it had StructuredTrace and Event, but then they got moved out to factory, and I went with the flow of adding generics everywhere 🤣